crates/codegen/runtime/cargo/src/runtime/bindings/mod.rs

```diff
@@ -1,10 +1,193 @@
 #[path = "generated/binding_rules.rs"]
 mod binding_rules;
 
-use metaslang_graph_builder::ast;
-pub use metaslang_graph_builder::functions::Functions;
-pub use metaslang_graph_builder::{ExecutionConfig, ExecutionError, NoCancellation, Variables};
+pub mod graph_builder {
+    use metaslang_graph_builder::{ast, functions, graph};
+
+    use crate::cst::KindTypes;
+
+    pub type File = ast::File<KindTypes>;
+    pub type Functions = functions::Functions<KindTypes>;
+    pub type Graph = graph::Graph<KindTypes>;
+
+    pub use metaslang_graph_builder::{ExecutionConfig, NoCancellation, Variables};
+}
+
+use std::collections::{BTreeSet, HashMap};
+use std::fmt;
+use std::fmt::Debug;
+use std::iter::once;
+use std::path::PathBuf;
+
+use metaslang_graph_builder::stack_graph;
+use semver::Version;
+use stack_graphs::graph::StackGraph;
+use stack_graphs::partial::PartialPaths;
+use stack_graphs::stitching::{ForwardPartialPathStitcher, GraphEdgeCandidates, StitcherConfig};
+use thiserror::Error;
 
 use crate::cst::KindTypes;
+use crate::cursor::Cursor;
+
+type Builder<'a> = stack_graph::Builder<'a, KindTypes>;
+type GraphHandle = stack_graphs::arena::Handle<stack_graphs::graph::Node>;
+
+#[derive(Error, Debug)]
+pub enum BindingsError {
+    #[error(transparent)]
+    IoError(#[from] std::io::Error),
+
+    #[error("Unknown error condition: {0}")]
```

Comment:
Two of these errors are unused, and the third is known. I wonder if this is outdated? is it used later?

Reply:
Both are used in the `get_globals_for_file`. But I agree they don't make much sense. And they could be collapsed together with `BuildError` below. I'll rework this and try to improve it.

Reply:
I removed both of these, as they were used when obtaining the `ROOT_PATH` from the given file path. Since we're no longer providing the `ROOT_PATH` global variable, they are not needed anymore.